pkg/controller/windowsmachineconfig/tracker.go

```diff
@@ -79,14 +75,110 @@ func newNodeRecord(ipAddress string, secretName string) nodeRecord {
+	}
+	encodedCreds := instanceSecret.Data[instanceID]
+	if err := json.Unmarshal(encodedCreds, &creds); err != nil {
+		return creds, errors.Wrap(err, "unmarshalling creds failed")
+	}
+	return creds, err
+}
+
+// initWindowsVMs initializes the windowsVMs map from the store configmMap and corresponding secrets
+func initWindowsVMs(k8sclientset *kubernetes.Clientset, operatorNS string) (map[types.WindowsVM]bool, error) {
+	windowsVMs := make(map[types.WindowsVM]bool)
+	store, err := k8sclientset.CoreV1().ConfigMaps(operatorNS).Get(StoreName, metav1.GetOptions{})
+	// It is ok if the tracker configmap doesn't exist. It is usually the case when the operator starts
+	// for the first time but it is not ok when the tracker configmap exists but it unqueryable for some reason
+	if err != nil && !k8serrors.IsNotFound(err) {
+		return nil, errors.Wrap(err, fmt.Sprintf("unable to query %s/%s ConfigMap", operatorNS, StoreName))
+	}
+	if err != nil && k8serrors.IsNotFound(err) {
+		log.Info("tracker config map doesn't exist yet so we can skip initialization of the WindowsVM map")
+		return windowsVMs, nil
+	}
+	// TODO: Get all the running Windows nodes in the cluster
+	//		jira story: https://issues.redhat.com/browse/WINC-280
+	nodeList, err := k8sclientset.CoreV1().Nodes().List(metav1.ListOptions{LabelSelector: nc.WindowsOSLabel})
+	if err != nil {
+		return nil, errors.Wrap(err, "error while querying for Windows nodes")
+	}
+	for instanceID := range store.BinaryData {
+		ipAddress, err := getInstanceIP(nodeList, instanceID)
+		if err != nil {
+			// As of now, we're doing best effort to reconstruct the configMap, so ignore errors while getting them
+			// from configMap. Having said that, the configMap entries should be perfect. Please look at syncNodeRecords
+			// method, where we disallow populating configMap with empty IPAddress, InstanceID and credentials to access
+			// the Windows VM
+			log.Error(err, "error getting external ip", "instance", instanceID)
+			continue
+		}
+		credentials, err := getCredsFromSecret(k8sclientset, instanceID, operatorNS)
+		if err != nil {
+			log.Error(err, "error getting credentials", "instance", instanceID)
+			continue
+		}
+		windowsVM, err := getWindowsVM(ipAddress, instanceID, credentials)
+		if err != nil {
+			log.Error(err, "error constructing windowsVM object", "instance", instanceID)
+			continue
+		}
+		// Insert the windowsVM created.
+		windowsVMs[windowsVM] = true
+	}
+	return windowsVMs, nil
+}
+
 // Reconcile ensures that the ConfigMap used by the tracker to store VM -> node information is present on the cluster
 // and is populated with nodeRecords.
-func (t *Tracker) Reconcile() error {
+func (t *tracker) Reconcile() error {
 	store, err := t.k8sclientset.CoreV1().ConfigMaps(t.operatorNS).Get(StoreName, metav1.GetOptions{})
 	if err != nil && !k8serrors.IsNotFound(err) {
 		return errors.Wrap(err, fmt.Sprintf("unable to query %s/%s ConfigMap", t.operatorNS, StoreName))
```

Comment:
can use errors.wrapf instead of fmt

Reply:
Beyond the scope of the PR but I can change it, if everyone else is fine.

Reply:
I agree with @ravisantoshgudimetla but I won't nack given the scope of the change.

Reply:
this is seen in multiple places, so we can let this be for now and I can change this as a part of my hack days for the repo.